shared-kernel-errors/src/main/java/org/dcsa/skernel/errors/infrastructure/SpringExceptionHandler.java

```diff
@@ -201,4 +205,33 @@ public ResponseEntity<RequestFailureTO> handleMissingServletRequestParameterExce
       new ConcreteRequestErrorMessageTO("invalidInput", e.getMessage())
     );
   }
+
+  @ExceptionHandler(HttpMediaTypeNotSupportedException.class)
+  public ResponseEntity<RequestFailureTO> handleHttpMediaTypeNotSupportedException(
+    HttpServletRequest httpServletRequest, HttpMediaTypeNotSupportedException e) {
+    log.debug("Exception {}: {}", e.getClass().getName(), e.getMessage());
+    return response(
+      httpServletRequest,
+      HttpStatus.UNSUPPORTED_MEDIA_TYPE,
+      new ConcreteRequestErrorMessageTO("unsupported media type", e.getMessage())
+    );
+  }
+
+  @ExceptionHandler(HttpRequestMethodNotSupportedException.class)
+  public ResponseEntity<RequestFailureTO> handleHttpRequestMethodNotSupportedException(
+    HttpServletRequest httpServletRequest, HttpRequestMethodNotSupportedException e) {
+    log.debug("Exception {}: {}", e.getClass().getName(), e.getMessage());
+    MultiValueMap<String, String> headers = null;
+    String[] methods = e.getSupportedMethods();
+    if (methods != null) {
+      headers = new HttpHeaders();
+      headers.add("Allow", String.join(", ", methods));
+    }
+    return response(
+      httpServletRequest,
+      HttpStatus.METHOD_NOT_ALLOWED,
+      headers,
+      new ConcreteRequestErrorMessageTO("method not allowed", e.getMessage())
+    );
+  }
```

Comment:
Looks like a duplicate of `MethodNotSupposedExceptionHandler.class`

Reply:
HttpRequestMethodNotSupportedException is a spring exception was but for some reason in it's own class, so moved it here for consistency